lib/jekyll/polyglot/patches/jekyll/site.rb

```diff
@@ -114,15 +114,14 @@ def derive_lang_from_path(doc)
       end
 
       segments = doc.relative_path.split('/')
-      if doc.relative_path[0] == '_' \
-        && segments.length > 2 \
-        && segments[1] =~ /^[a-z]{2,3}(:?[_-](:?[A-Za-z]{2}){1,2}){0,2}$/
-        return segments[1]
-      elsif segments.length > 1 \
-        && segments[0] =~ /^[a-z]{2,3}(:?[_-](:?[A-Za-z]{2}){1,2}){0,2}$/
-        return segments[0]
-      end
 
+      # loop through all segments and check if they match the language regex
+      segments.each do |segment|
+        if segment =~ /^[a-z]{2,3}(:?[_-](:?[A-Za-z]{2}){1,2}){0,2}$/
```

Comment:
Thank you @george-gca for another contribution! sorry for the belated response.

I do want raised standards on asking for test changes with PRs, but honestly this project needs help with testing in general. Jekyll plugins are complex to test, and the framework I have setup now is pretty barebones but it works and we can call this function from the test spec. I can help with testing this PR.

I like this approach because it's very elegant code wise, and I can see the benefit of this feature. But  I'm cringing looking at this PR because my past code brings me dishonor; I have pointless copypasta on a regex that's greedy and entirely unnecessary.

lets make this even cleaner, yeah?

```suggestion
        if languages.include?(segment)
```

Reply:
I thought this regex had a more special need that I didn't know about haha. Sure.